### PR TITLE
changing verbiage of AKS pod scheduling on worker nodes

### PR DESCRIPTION
**Pull request by Dost2010**
# Overview/Summary

Changing verbiage of AKS pod scheduling on worker nodes

## Related Issues/Work Items

Replace this with a list of related GitHub Issues and/or ADO Work Items (Internal Only)

- To associate a GitHub Issue, use a [key word](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) preceded with the GitHub issue number.
- To associate an ADO Work Item, use the key word `AB#` succeeded with the [ADO Work Item ID](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## This PR fixes/adds/changes/removes

1. Changing verbiage of AKS pod scheduling on worker nodes

### Breaking Changes

None 

## As part of this pull request I have

- [ ] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [ ] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [ ] Ensured PR tests are passing
- [ ] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [ ] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

File: azure-resources/ContainerService/managedClusters/recommendations.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@
   recommendationResourceType: Microsoft.ContainerService/managedClusters
   recommendationMetadataState: Active
   longDescription: |
-    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling system pods should be scheduled there.
+    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling only system pods should be scheduled there. For worker nodes assign taints and label CriticalAddonsOnly to node.
```
Comment (oZakari) [inline azure-resources/ContainerService/managedClusters/recommendations.yaml:30]:
```suggestion
    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling the preference for system pods should be scheduled there. The CriticalAddonsOnly=true:NoSchedule taint can be added to your system nodes to enforce the prevention of application pods from being scheduled on them. 
```

Comment (oZakari) [reply azure-resources/ContainerService/managedClusters/recommendations.yaml:30]: Adjust the wording as the system label doesn't explicitly block application pods from being added to those system nodes. Also adding clarity to use same terminology referenced in docs around system nodes.

**Review by oZakari — CHANGES_REQUESTED** [review]

File: azure-resources/ContainerService/managedClusters/recommendations.yaml (hunk 1 above)
Comment (Dost2010) [reply azure-resources/ContainerService/managedClusters/recommendations.yaml:30]: Good with it the new wording and labeling :) LTGM

Comment (Dost2010) [reply azure-resources/ContainerService/managedClusters/recommendations.yaml]: I have approved the commit but it seems that the number of character is more than 300. Can you please readjust few characters in your suggestion to make it comply with the description.

File: azure-resources/ContainerService/managedClusters/recommendations.yaml [hunk 2]
```diff
@@ -27,7 +27,7 @@
   recommendationResourceType: Microsoft.ContainerService/managedClusters
   recommendationMetadataState: Active
   longDescription: |
-    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling system pods should be scheduled there.
+    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling the preference for system pods should be scheduled there. The CriticalAddonsOnly=true:NoSchedule taint can be added to your system nodes to enforce the prevention of application pods from being scheduled on them.
```
Comment (oZakari) [inline azure-resources/ContainerService/managedClusters/recommendations.yaml:30]:
```suggestion
    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling the preference for system pods should be scheduled there. The CriticalAddonsOnly=true:NoSchedule taint can be added to your system nodes to prohibit application pods from being scheduled on them. 
```

**Review by oZakari — APPROVED** [review]

File: azure-resources/ContainerService/managedClusters/recommendations.yaml [hunk 3]
```diff
@@ -27,7 +27,7 @@
   recommendationResourceType: Microsoft.ContainerService/managedClusters
   recommendationMetadataState: Active
   longDescription: |
-    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling system pods should be scheduled there.
+    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling the preference for system pods should be scheduled there. The CriticalAddonsOnly=true:NoSchedule taint can be added to your system nodes to prohibit application pods from being scheduled on them.
```
Comment (oZakari) [inline azure-resources/ContainerService/managedClusters/recommendations.yaml:30]:
```suggestion
    AKS assigns the kubernetes.azure.com/mode: system label to nodes in system node pools signaling the preference for system pods should be scheduled there. The CriticalAddonsOnly=true:NoSchedule taint can be added to your system nodes to prohibit application pods from being scheduled on them.
```

**Review by oZakari — APPROVED** [review]

**Review by ejhenry — APPROVED** [review]

**Pull request merged**